src/shared/generator.c

```diff
@@ -138,6 +138,13 @@ int generator_write_fsck_deps(
         assert(what);
         assert(where);
 
+        /* Let's do an early exit if we are invoked for the root and /usr/ trees in the initrd, to avoid
+         * generating confusing log messages */
+        if (in_initrd() && PATH_IN_SET(where, "/", "/usr")) {
+                log_debug("Skipping fsck for / + /usr in initrd.");
```

Comment:
Maybe `("Skipping fsck for %s in initrd.", path)` ? The message is not very clear right now.